File: internal/dto/container_dto.go
package dto

type CreateContainerRequest struct {
	Name          string  `json:"name" binding:"required,min=2,max=100"`
	InnerLengthMM float64 `json:"inner_length_mm" binding:"required,gt=0"`
	InnerWidthMM  float64 `json:"inner_width_mm" binding:"required,gt=0"`
	InnerHeightMM float64 `json:"inner_height_mm" binding:"required,gt=0"`
	MaxWeightKG   float64 `json:"max_weight_kg" binding:"required,gt=0"`
	Description   *string `json:"description" binding:"omitempty,max=500"`
}

type UpdateContainerRequest struct {
	Name          string  `json:"name" binding:"required,min=2,max=100"`
	InnerLengthMM float64 `json:"inner_length_mm" binding:"required,gt=0"`
	InnerWidthMM  float64 `json:"inner_width_mm" binding:"required,gt=0"`
	InnerHeightMM float64 `json:"inner_height_mm" binding:"required,gt=0"`
	MaxWeightKG   float64 `json:"max_weight_kg" binding:"required,gt=0"`
	Description   *string `json:"description" binding:"omitempty,max=500"`
}

type ContainerResponse struct {
	ID            string  `json:"id"`
	Name          string  `json:"name"`
	InnerLengthMM float64 `json:"inner_length_mm"`
	InnerWidthMM  float64 `json:"inner_width_mm"`
	InnerHeightMM float64 `json:"inner_height_mm"`
	MaxWeightKG   float64 `json:"max_weight_kg"`
	Description   *string `json:"description,omitempty"`
}

File: mobile/lib/dtos/container_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'container_dto.g.dart';

@JsonSerializable()
class ContainerResponseDto {
  final String id;
  final String name;
  @JsonKey(name: 'inner_length_mm')
  final double innerLengthMm;
  @JsonKey(name: 'inner_width_mm')
  final double innerWidthMm;
  @JsonKey(name: 'inner_height_mm')
  final double innerHeightMm;
  @JsonKey(name: 'max_weight_kg')
  final double maxWeightKg;
  final String? description;

  ContainerResponseDto({
    required this.id,
    required this.name,
    required this.innerLengthMm,
    required this.innerWidthMm,
    required this.innerHeightMm,
    required this.maxWeightKg,
    this.description,
  });

  factory ContainerResponseDto.fromJson(Map<String, dynamic> json) => _$ContainerResponseDtoFromJson(json);
}

@JsonSerializable()
class CreateContainerRequestDto {
  final String name;
  @JsonKey(name: 'inner_length_mm')
  final double innerLengthMm;
  @JsonKey(name: 'inner_width_mm')
  final double innerWidthMm;
  @JsonKey(name: 'inner_height_mm')
  final double innerHeightMm;
  @JsonKey(name: 'max_weight_kg')
  final double maxWeightKg;
  final String? description;

  CreateContainerRequestDto({
    required this.name,
    required this.innerLengthMm,
    required this.innerWidthMm,
    required this.innerHeightMm,
    required this.maxWeightKg,
    this.description,
  });

  Map<String, dynamic> toJson() => _$CreateContainerRequestDtoToJson(this);
}

@JsonSerializable()
class UpdateContainerRequestDto {
  final String name;
  @JsonKey(name: 'inner_length_mm')
  final double innerLengthMm;
  @JsonKey(name: 'inner_width_mm')
  final double innerWidthMm;
  @JsonKey(name: 'inner_height_mm')
  final double innerHeightMm;
  @JsonKey(name: 'max_weight_kg')
  final double maxWeightKg;
  final String? description;

  UpdateContainerRequestDto({
    required this.name,
    required this.innerLengthMm,
    required this.innerWidthMm,
    required this.innerHeightMm,
    required this.maxWeightKg,
    this.description,
  });

  Map<String, dynamic> toJson() => _$UpdateContainerRequestDtoToJson(this);
}
